src/communication/v2/pool.hpp

```diff
@@ -22,42 +22,32 @@
 
 namespace memgraph::communication::v2 {
 
-class IOContextPool {
+class IOContextThreadPool {
  private:
   using IOContext = boost::asio::io_context;
   using IOContextGuard = boost::asio::executor_work_guard<boost::asio::io_context::executor_type>;
 
  public:
-  explicit IOContextPool(size_t pool_size) : next_io_context_(0) {
+  explicit IOContextThreadPool(size_t pool_size) : guard_{io_context_.get_executor()}, pool_size_{pool_size} {
     MG_ASSERT(pool_size != 0, "Pool size must be bigger then 0!");
-
-    // Give all the io_contexts work to do so that their Run() functions will not
-    // exit until they are explicitly stopped.
-    for (size_t i = 0; i < pool_size; ++i) {
-      io_contexts_.emplace_back(std::make_shared<IOContext>());
-      guards_.emplace_back(std::make_shared<IOContextGuard>(io_contexts_.back()->get_executor()));
-    }
   }
 
-  IOContextPool(const IOContextPool &) = delete;
-  IOContextPool &operator=(const IOContextPool &) = delete;
-  IOContextPool(IOContextPool &&) = delete;
-  IOContextPool &operator=(IOContextPool &&) = delete;
-  ~IOContextPool() { MG_ASSERT(background_threads_.empty(), "Error while destructing thread pool"); };
+  IOContextThreadPool(const IOContextThreadPool &) = delete;
+  IOContextThreadPool &operator=(const IOContextThreadPool &) = delete;
+  IOContextThreadPool(IOContextThreadPool &&) = delete;
+  IOContextThreadPool &operator=(IOContextThreadPool &&) = delete;
+  ~IOContextThreadPool() { MG_ASSERT(background_threads_.empty(), "Error while destructing context thread pool"); };
```

Comment:
What is the reason for this assert? I think in an RAII class the destructor should take care of this. A good example is how the standard now contains `std::jthread`.